core/string/ustring.cpp

```diff
@@ -3314,6 +3314,10 @@ int String::find(const String &p_str, int p_from) const {
 		return -1; // won't find anything!
 	}
 
+	if (src_len == 1) {
+		return find_char(p_str[0], p_from); // optimize with single-char find
+	}
```

Comment:
Love that this will optimize even `String::find()` and `String::contains()` calls where the string is one char long.